library/src/interactor_impl.cxx

```diff
@@ -268,6 +282,88 @@ class interactor_impl::internals
     self->Window.render();
   }
 
+  static void OnMiddleButtonPress(vtkObject*, unsigned long, void* clientData, void*)
+  {
+    internals* self = static_cast<internals*>(clientData);
+
+    const auto click_pos = self->VTKInteractor->GetEventPosition();
+    self->MiddleClickPosision[0] = click_pos[0];
+    self->MiddleClickPosision[1] = click_pos[1];
+
+    self->Style->OnMiddleButtonDown();
+  }
+
+  static void OnMiddleButtonRelease(vtkObject*, unsigned long, void* clientData, void*)
+  {
+    internals* self = static_cast<internals*>(clientData);
+
+    const auto click_pos = self->VTKInteractor->GetEventPosition();
```

Comment:
Avoid `auto` for simple types. I'm not against it, but we try to be consistent with the existing code style.

Reply:
I kept `auto` for the `std::chrono::` stuff that gets quite verbose, let me know if I should change that too.